src/components/SettlementButton.tsx

```diff
@@ -1,143 +1,106 @@
-import PropTypes from 'prop-types';
-import React, {useEffect, useMemo} from 'react';
-import {withOnyx} from 'react-native-onyx';
-import _ from 'underscore';
+import _ from 'lodash';
+import React, {MutableRefObject, useEffect, useMemo} from 'react';
+import {StyleProp, ViewStyle} from 'react-native';
+import {OnyxEntry, withOnyx} from 'react-native-onyx';
 import useLocalize from '@hooks/useLocalize';
 import useNetwork from '@hooks/useNetwork';
-import compose from '@libs/compose';
 import * as ReportUtils from '@libs/ReportUtils';
-import iouReportPropTypes from '@pages/iouReportPropTypes';
 import * as BankAccounts from '@userActions/BankAccounts';
 import * as IOU from '@userActions/IOU';
 import * as PaymentMethods from '@userActions/PaymentMethods';
 import CONST from '@src/CONST';
 import ONYXKEYS from '@src/ONYXKEYS';
 import ROUTES from '@src/ROUTES';
+import {ButtonSizeValue} from '@src/styles/utils/types';
+import type {Report} from '@src/types/onyx';
+import {EmptyObject} from '@src/types/utils/EmptyObject';
 import ButtonWithDropdownMenu from './ButtonWithDropdownMenu';
 import * as Expensicons from './Icon/Expensicons';
 import KYCWall from './KYCWall';
-import withNavigation from './withNavigation';
 
-const propTypes = {
-    /** Callback to execute when this button is pressed. Receives a single payment type argument. */
-    onPress: PropTypes.func.isRequired,
+type AnchorAlignment = {
```

Comment:
this type already exists in [Popover](https://github.com/Expensify/App/blob/main/src/components/Popover/types.ts) component , please reuse it